luigi/task.py

```diff
@@ -575,6 +575,40 @@ def output(self):
         """
         return []  # default impl
 
+    def process_requires(self, requires):
+        """
+        Override in "template" tasks which themselves are supposed to be
+        subclassed and thus have their requires() overridden (name preserved to
+        provide consistent end-user experience), yet need to introduce
+        custom behaviour into dependencies.
+
+        Differences with :py:meth:`_requires`:
+          * method signature: :py:meth:`_requires` doesn't take params and calls
+            :py:meth:`requires` directly where as this method is expected to work only on
+            requirements passed via parameters.
+          * Code paths: :py:meth:`_requires` is run only during the statically defined
+            requirements where as this method works with dynamic dependencies yielded from
+            :py:meth:`run` also.
+
+        ex:
+            class ParentTask(luigi.Task):
+                foo = `bar`
+
+                def process_requires(self, requirements):
+                    for(req in flatten(requirements)):
+                        req.foo = self.foo
+
+                    return requirements
+
+            class B(ParentTask):
+                def requires(self):
+                    return luigi.LocalPathTask('my/local/path')
+
+            B().deps()[0].foo # => 'bar'
```

Comment:
I know `deps()` isn't prefixed with `_` but I think we would like to make it private, so better not referencing it either.

Reply:
We could remove this. I'm open to ideas on how we can demonstrate the results of the override?

Reply:
I think it's fine to just remove this line for now. The behavior is already described in text above right?